projects/ngx-d3timeline/src/lib/scale.service.ts

```diff
@@ -0,0 +1,66 @@
+import { scaleBand, ScaleBand, scaleTime, ScaleTime } from 'd3-scale';
+import { min, max } from 'd3-array';
+import { TimelineEvent } from './timeline-event';
+import { TimelineView } from './view/timeline-view';
+import { Orientation } from './orientation';
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class ScaleService {
```

Comment:
I am wondering if both scales should be in the same service? There is quite a bit going on in this one...

There are 2 functions which do very similar things but differ in what scale they apply to, getScaleBandDomain and getScaleTimeDomain. Similar with the configure methods.

It looks like there is only a single common function, getRange()

Reply:
done